CrisisAdjustment/crisis_deaths_to_qx.R

```diff
@@ -0,0 +1,233 @@
+            "_shp/", country, "_Amat_Names.rda"))
+
+# UN-IGME national crisis adjustments
+igme <- readxl::read_xlsx("CrisisAdjustment/Crisis_Under5_deaths_2022.xlsx")
+
+
+# Prep population from UN-IGME pop and weights ----------------------------
+
+# national pop
+pop <- igme[igme$Country == country, c("Country", "Year", "Pop 0", "Pop 1-4")]
+names(pop) <- c("country", "years", "nat_pop_0_1", "nat_pop_1_5")
+pop$years <- as.integer(floor(pop$years))
+
+# Admin 1: merge on weights and format
+pop_adm1 <- merge(weight.adm1.u1, weight.adm1.u5, by = c("region", "years"))
+names(pop_adm1) <- c("region", "years", "prop_0_1", "prop_1_5")
+pop_adm1$level <- "admin1"
+pop_adm1 <- merge(pop_adm1, admin1.names, by.x = "region", by.y = "Internal")
+names(pop_adm1)[names(pop_adm1) == "GADM"] <- "gadm"
+pop_adm1 <- merge(pop_adm1, pop, by = "years")
+
+# Admin 2: merge on weights and format
+pop_adm2 <- merge(weight.adm2.u1, weight.adm2.u5, by = c("region", "years"))
+names(pop_adm2) <- c("region", "years", "prop_0_1", "prop_1_5")
+pop_adm2$level <- "admin2"
+pop_adm2 <- merge(pop_adm2, admin2.names, by.x = "region", by.y = "Internal")
+names(pop_adm2)[names(pop_adm2) == "GADM"] <- "gadm"
+pop_adm2 <- merge(pop_adm2, pop, by = "years")
+
+# combine and compute area-level population from weights and national pop
+pop <- rbind(pop_adm1, pop_adm2)
+pop$pop_0_1 <- pop$prop_0_1 * pop$nat_pop_0_1
+pop$pop_1_5 <- pop$prop_1_5 * pop$nat_pop_1_5
+
+# final columns: country, level (i.e. admin1, admin2), gadm (area name),
+# region (area code), years, pop_0_1, pop_1_5
+pop <- pop[, c("country", "level", "gadm", "region", "years", "pop_0_1", "pop_1_5")]
+
+
+# utilities ----------------------------------------------------------------
+
+# Function for crisis-specific 5q0  from u1 and 1to4 deaths and population
+# Input: df [data.frame; columns listed below]
+# - ed_0_1: excess deaths age 0 years
+# - ed_1_5: excess deaths age 1-4 years
+# - pop_0_1: population age 0 years
+# - pop_1_5: population age 1-4 years
+get_ed_5q0 <- function (df) {
+  df <- df %>%
+    mutate(ed_1m0 = ed_0_1 / pop_0_1,
+           ed_4m1 = ed_1_5 / pop_1_5) %>%
+    mutate(ed_1q0 = ed_1m0 / (1 + (1 - (0.3)) * ed_1m0),
+           ed_4q1 = 4 * ed_4m1 / (1 + (4 - 4 * (0.4)) * ed_4m1)) %>%
+    mutate(ed_5q0 = 1 - (1 - ed_1q0) * (1 - ed_4q1))
+  return(df)
+}
+
+
+# Special cases -----------------------------------------------------------
+
+# Liberia only has admin1 ebola deaths; split to admin2 by pop
+if (country == "Liberia") {
+  
+  # create empty starting point
+  deaths_adm2 <- data.frame(
+    country = "Liberia", level = "admin2",
+    gadm = admin2.names$GADM,
+    region = admin2.names$Internal
+  )
+  gadm <- rgdal::readOGR(
+    dsn = "CrisisAdjustment/gadm41_LBR_shp",
+    layer = "gadm41_LBR_2"
+  )@data[, c("NAME_1", "NAME_2")]
+  # TODO: does Liberia have multiple admin2 with same name? If yes the following 2 merges will be bad.
```

Comment:
@alanamcgovern Do you remember how we fixed this issue in the past? The merge needs to happen on the region ID (e.g. `admin2_4` style) but I don't remember how those variables are generated onto the `gadm` object here.

Reply:
I can fix that.